src/python/controller/extract/extract.py

```
@@ -128,124 +117,105 @@ def is_archive_fast(archive_path: str) -> bool:
+                if os.path.islink(os.path.join(dirpath, name)):
+                    os.remove(os.path.join(dirpath, name))
+                    continue
+                try:
+                    common = os.path.commonpath([real_out_dir, full_path])
+                except ValueError:
+                    common = None
+                if common != real_out_dir:
+                    raise ExtractError(
+                        "Zip-slip detected: extracted path '{}' escapes target directory '{}'".format(
+                            full_path, real_out_dir
                         )
```

Ruff (0.15.4)

[warning] Use f-string instead of `format` call

Convert to f-string

(UP032)